### PR TITLE
Alex/feature/edit account template

### PR DESCRIPTION
**Pull request by AlexandrBelenov**
I've developed account extension logic.

### CODE REVIEW

File: app/views/account_extension.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        return redirect(url_for('main.accounts'))
+    form = AccountExtensionForm(
+        id=extension.id,
+        product_id=extension.product_id,
+        reseller_id=extension.reseller_id,
+        months=extension.months,
+        extension_date=extension.extension_date
+        )
+    form.is_edit = True
+    form.products = Product.query.filter(Product.deleted == False) # noqa E712
```
Comment (kipod) [inline app/views/account_extension.py]: .all()

File: app/views/account_extension.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    form.resellers = Reseller.query.filter(Reseller.deleted == False)  # noqa E712
```
Comment (kipod) [inline app/views/account_extension.py]: ... .all()

File: app/views/reseller_product.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+@reseller_product_blueprint.route("/reseller_product_delete", methods=['GET'])
+@login_required
+def delete():
+    log(log.INFO, '%s /reseller_product_remove', request.method)
+    if 'id' not in request.args:
+        flash('Unknown reseller product id', 'danger')
+        return redirect(url_for('main.resellers'))
+    product = ResellerProduct.query.filter(ResellerProduct.id == int(request.args['id'])).first()
+    reseller_id = product.reseller_id
+    ninja_product = ninja.get_product(product.ninja_product_id)
```
Comment (kipod) [inline app/views/reseller_product.py:37]: need check if ninja unreachable

File: app/views/reseller_product.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    log(log.INFO, '/reseller_product_save')
+    form = ResellerProductForm(request.form)
+    if not form.validate_on_submit():
+        flash('Form validation error', 'danger')
+        log(log.WARNING, "Form validation error %s", form.errors)
+        return redirect(url_for('reseller.edit', id=form.reseller_id.data))
+    # product = ResellerProduct.query\
+    #     .filter(ResellerProduct.reseller_id == form.reseller_id.data)\
+    #     .filter(ResellerProduct.product_id == form.product_id.data)\
+    #     .filter(ResellerProduct.months == form.months.data).first()
```
Comment (kipod) [inline app/views/reseller_product.py]: pls remove

File: flask_app.py [hunk 5]
```diff
@@ -41,20 +42,20 @@ def add_phone(name, price):
...
-    add_reseller_product(months=1, price=4.25, product=product_silver, reseller=reseller_nitrix)
-    add_reseller_product(months=3, price=11.50, product=product_silver, reseller=reseller_nitrix)
-    add_reseller_product(months=6, price=19.45, product=product_silver, reseller=reseller_nitrix)
-    add_reseller_product(months=12, price=35.99, product=product_silver, reseller=reseller_nitrix)
-    add_reseller_product(months=1, price=2.50, product=product_bronsa, reseller=reseller_nitrix)
-    add_reseller_product(months=3, price=6.99, product=product_bronsa, reseller=reseller_nitrix)
-    add_reseller_product(months=6, price=12.45, product=product_bronsa, reseller=reseller_nitrix)
-    add_reseller_product(months=12, price=20.99, product=product_bronsa, reseller=reseller_nitrix)
+    # product_silver = Product(name='Silver').save(False)
+    # product_bronsa = Product(name='Bronsa').save(False)
```
Comment (kipod) [inline flask_app.py]: need remove

File: flask_app.py [hunk 6]
```diff
@@ -41,20 +42,20 @@ def add_phone(name, price):
...
+    add_reseller_product(months=6, initprice=30.45, extprice=19.55, product=product_gold, reseller=reseller_nitrix)
+    # add_reseller_product(months=12, price=50.0, product=product_gold, reseller=reseller_nitrix)
+    # add_reseller_product(months=1, price=4.25, product=product_silver, reseller=reseller_nitrix)
+    # add_reseller_product(months=3, price=11.50, product=product_silver, reseller=reseller_nitrix)
+    # add_reseller_product(months=6, price=19.45, product=product_silver, reseller=reseller_nitrix)
+    # add_reseller_product(months=12, price=35.99, product=product_silver, reseller=reseller_nitrix)
+    # add_reseller_product(months=1, price=2.50, product=product_bronsa, reseller=reseller_nitrix)
+    # add_reseller_product(months=3, price=6.99, product=product_bronsa, reseller=reseller_nitrix)
+    # add_reseller_product(months=6, price=12.45, product=product_bronsa, reseller=reseller_nitrix)
+    # add_reseller_product(months=12, price=20.99, product=product_bronsa, reseller=reseller_nitrix)
```
Comment (kipod) [inline flask_app.py]: need remove

File: tests/test_reseller_product.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        db.session.remove()
+        db.drop_all()
+        app_ctx.pop()
+
+
+def test_edit_reseller(client):
+    TEST_PROD_ID = 3
+    reseller_product = ResellerProduct(product_id=TEST_PROD_ID)
+    reseller_product.save()
+    response = client.get(f'/reseller_product_edit?id={reseller_product.id}')
```
Comment (kipod) [inline tests/test_reseller_product.py]: use here url_for

File: tests/test_reseller_product.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    assert response.status_code == 200
+    assert b'3' in response.data
```
Comment (kipod) [inline tests/test_reseller_product.py]: need add tests for:
* url_for('reseller_product.add')
* url_for('reseller_product.delete')
* url_for('reseller_product.edit')
* url_for('reseller_product.save')

**Review by kipod — COMMENTED** [review]
Thanks, guys!
Good job.
Please fix small issues and push this branch again.
Tests we need to provide also, but then can be done in a separate task.

File: app/views/reseller_product.py (hunk 3 above)
Comment (AlexandrBelenov) [reply app/views/reseller_product.py:37]: I've added if...else block to check whether a ninja product'd been retrieved

**Comment by kipod** [comment]
не замержил с девелоп бранчем ;)

**Pull request merged**